pywbem/_subscription_manager.py

```diff
@@ -1162,35 +1176,37 @@ def _create_destination(self, server_id, dest_url, owned):
         # Validate server_id
         server = self._get_server(server_id)
 
+        this_client = getfqdn()
+
         # Validate the URL by reconstructing it. Do not allow defaults
         _, _, listener_url = parse_url(dest_url, allow_defaults=False)
 
-        this_host = getfqdn()
         ownership = "owned" if owned else "permanent"
 
-        dest_path = CIMInstanceName(DESTINATION_CLASSNAME,
-                                    namespace=server.interop_ns)
-
         dest_inst = CIMInstance(DESTINATION_CLASSNAME)
-        dest_inst.path = dest_path
         dest_inst['CreationClassName'] = DESTINATION_CLASSNAME
         dest_inst['SystemCreationClassName'] = SYSTEM_CREATION_CLASSNAME
-        dest_inst['SystemName'] = this_host
+        dest_inst['SystemName'] = self._systemnames[server_id]
+
         dest_inst['Name'] = _format(
-            'pywbemdestination:{0}:{1}:{2}',
-            ownership, self._subscription_manager_id, uuid.uuid4())
+            'pywbemdestination:{0}:{1}:{2}:{3}',
+            ownership, this_client, self._subscription_manager_id, uuid.uuid4())
         dest_inst['Destination'] = listener_url
 
         if owned:
+            dest_path = self._build_path(dest_inst,
+                                         server.interop_ns,
+                                         server_id)
             for i, inst in enumerate(self._owned_destinations[server_id]):
-                if inst.path == dest_path:
+                if self._match_path(inst.path, dest_path):
                     # It already exists, now check its properties
                     if inst != dest_inst:
```

Comment:
We should discuss whether modifying the instance in case we identified it but its properties are not as desired, is such a good idea after all. It bears the risk that this changes something that was intentionally done that way before.

Reply:
Agreed but I want to think it through with some examples first.